### PR TITLE
output-complete-obj: link runtime C libraries when building shared libraries

### PR DESCRIPTION
**Pull request by Octachron**
Thus PR is an offspring of #1351  : currently, when building a shared library with `-output-obj` or `-output-complete-obj`:
```
ocamlopt -output-obj a.ml -o a.so 
```
the only difference between `-output-obj` and `output-complete-obj` is that `-output-complete-obj` does *not* link the C libraries used by the runtime by default. The fact that the `complete` variant links less objects than the simple variant is not intuitive at all — as illustrated by the discussion in #1351 . 

Moreover, @whitequark confirmed that this behavior was not really intended and was inherited from the static linking mode.

Consequently, this small PR proposes to align the behavior of `-output-complete-obj` to the behavior of `output-obj` and link these runtime C libraries in the shared library mode.

### CODE REVIEW

**Comment by xavierleroy** [comment]
18 months later, is this PR still relevant?  @Octachron, you're in control, do what it takes.

**Comment by gasche** [comment]
It looks to me like this PR is still relevant and has not been invalidated by recent developments around `-custom`, etc., but we need a review from someone that knows about this stuff.

(cc @nojb ?)

**Comment by nojb** [comment]
I had taken a look at this some time back and it seemed OK. I will do a formal review shortly.

**Review by nojb — APPROVED** [review]
Looks good, thanks! Please rebase it so that it can be merged.

**Comment by Octachron** [comment]
I somehow completely missed this ping, sorry! I updated the Change entry and will merge (once the CI is green).

**Pull request merged**
